app/models/manageiq/providers/openstack/cloud_manager/refresh_parser.rb

```diff
@@ -406,8 +406,8 @@ def clean_up_extra_flavor_keys
       end
     end
 
-    def add_instance_disk(disks, size, location, name)
-      super(disks, size, location, name, "openstack")
+    def add_instance_disk(disks, size, location, name, controller_type)
+      super(disks, size, location, name, controller_type)
```

Comment:
please take a look at the test we have the same locations for many disks and we have hardcoded controller_type which makes openstack disks not unique. I had to change it.